modules/launcher/src/main/java/org/mule/module/launcher/DeploymentDirectoryWatcher.java

```diff
@@ -522,20 +522,26 @@ private void deleteAnchorsFromDirectory(final File directory)
 
     private void redeployModifiedDomains()
     {
-        redeployModifiedArtifacts(domains, domainTimestampListener, domainArchiveDeployer);
+        Collection redeployableDomains = getArtifactsToRedeploy(domains);
+        redeployModifiedArtifacts(redeployableDomains, domainTimestampListener, domainArchiveDeployer);
```

Comment:
Yes, any combination should be valid. It would be up to the user to take advantage of this. By default all domains will redeploy though so the past behaviour is guaranteed.